### PR TITLE
contrib: CERN plugin fixes and testing

### PR DESCRIPTION
**Pull request by omelkonian**
* CERN third-party authentication now completely working.

* Increases test coverage.

Signed-off-by: Orestis Melkonian <melkon.or@gmail.com>
Co-Authored-By: Esteban J. G. Gabancho <esteban.gabancho@gmail.com>

### CODE REVIEW

File: invenio_oauthclient/contrib/cern.py [hunk 1]
```diff
@@ -175,6 +179,7 @@ def fetch_groups(groups):
     return groups
 
 
+# TODO: Add cache for the dict
```
Comment (lnielsen) [inline invenio_oauthclient/contrib/cern.py]: TODO?

File: invenio_oauthclient/contrib/cern.py [hunk 2]
```diff
@@ -193,12 +198,35 @@ def account_info(remote, resp):
...
+    return dict(email=email.lower(),
+                profile=dict(nickname=nice, full_name=name),
+                external_id=external_id, external_method='cern',
+                active=True)
 
 
 def account_setup(remote, token, resp):
     """Perform additional setup after user have been logged in."""
-    pass
+    from invenio_oauthclient.utils import oauth_link_external_id
```
Comment (lnielsen) [inline invenio_oauthclient/contrib/cern.py]: Is it not possible to move the imports outside the function?

**Comment by omelkonian** [comment]
you can close Esteban's PR

File: examples/cern_app.py [hunk 3]
```diff
@@ -155,4 +174,5 @@ def index():
     if not current_user.is_authenticated:
         return redirect(url_for("invenio_oauthclient.login",
                                 remote_app='cern'))
+    # print(g.identity.__dict__)
```
Comment (egabancho) [inline examples/cern_app.py]: Probably you can remove this line 😉

File: examples/cern_app.py [hunk 4]
```diff
@@ -155,4 +152,8 @@ def index():
     if not current_user.is_authenticated:
         return redirect(url_for("invenio_oauthclient.login",
                                 remote_app='cern'))
-    return "hello {}".format(current_user.email)
+
+    display = "hello {}".format(current_user.email)
+    for group in [need.value for need in sorted(g.identity.provides)]:
```
Comment (egabancho) [inline examples/cern_app.py]: This is ok for testing purposes but I think we should remove it from here

**Comment by egabancho** [comment]
LGTM @lnielsen?

**Pull request merged**